crates/block-producer/src/runner.rs

```diff
@@ -338,3 +348,54 @@ pub fn run(config: Config) -> Result<()> {
 
     Ok(())
 }
+
+fn check_ckb_version(rpc_client: &RPCClient) -> Result<()> {
+    let ckb_version = block_on(rpc_client.get_ckb_version())?;
+    let ckb_version = ckb_version.split('(').collect::<Vec<&str>>()[0].trim_end();
+    if Version::parse(&ckb_version)? < Version::parse(MIN_CKB_VERSION)? {
+        return Err(anyhow!(
+            "The version of CKB node {} is lower than the minimum version {}",
+            ckb_version,
+            MIN_CKB_VERSION
+        ));
+    }
+    Ok(())
+}
+
+fn check_rollup_config_cell(
+    block_producer_config: &BlockProducerConfig,
+    rollup_config: &RollupConfig,
+    rpc_client: &RPCClient,
+) -> Result<()> {
+    let rollup_config_cell = block_on(
+        rpc_client.get_cell(
+            block_producer_config
+                .rollup_config_cell_dep
+                .out_point
+                .clone()
+                .into(),
+        ),
+    )?
+    .ok_or_else(|| anyhow!("can't find rollup config cell"))?;
+    let cell_data = format!("{:?}", rollup_config_cell.data.to_vec());
```

Comment:
1. We need to check that every `allowed_eoa_type_hashes` in the rollup config cell in is our godwoken config.
2. Parse the rollup config cell then do the comparation.